src/core/backend/pearl/core.rs

```diff
@@ -79,14 +81,48 @@ impl Pearl {
 #[async_trait]
 impl BackendStorage for Pearl {
     async fn run_backend(&self) -> Result<()> {
+        use futures::future::join_all;
```

Comment:
I think it's better to use tokio::spawn and wait on JoinHandlers to check result. I'm not sure that join_all provides true threaded parallelism. You can create generic init func, eg `init_groups(group: Vec<Group>, ...)`, and spawn it for each bunch